### PR TITLE
Full sonobuoy logs

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- Upload full sonobuoy logs (includes pod logs) until we have improved logs via OpenSearch. These changes can be reverted once that feature lands.
- Also upload logs with csp name included in artifact name so that there is one artifact per CSP.

### CODE REVIEW

**Comment by katexochen** [comment]
Do you have a test run for this change?

File: .github/actions/e2e_sonobuoy/action.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ name: sonobuoy
 description: "Execute the e2e test framework sonobuoy."
 
 inputs:
+  cloudProvider:
+    description: "Necessary for log naming."
```
Comment (malt3) [inline .github/actions/e2e_sonobuoy/action.yml:6]: nit: I would add the subject to the description. But maybe that's subjective so feel free to ignore.

**Review by malt3 — COMMENTED** [review]
Changeset looks fine. I would also like to see a test run.

**Comment by malt3** [comment]
> Now we do: https://github.com/edgelesssys/constellation/actions/runs/3620680468

The logs seem to be encoded twice (tar.gz from sonobuoy, zip from gh actions). That's not ideal but fine for me as we want to move towards centralized log collection anyways.

**Pull request merged**